roles/rsyslog/roles/output_roles/files_forwards/defaults/main.yaml

```diff
@@ -0,0 +1,193 @@
+---
+# Files and forwards output configuration
+# ---------------------------------
+
+# Files and Forwards Rsyslog output configuration rules
+# ---------------------------------
+rsyslog_files_and_forwards_output_rules:
+  - '{{ rsyslog_conf_files_output_modules }}'
+  - '{{ rsyslog_conf_files_output_rules }}'
+  - '{{ rsyslog_conf_forwards_output_rules }}'
+  - '{{ rsyslog_conf_remote_templates }}'
+
+# rsyslog_conf_files_output_modules:
+rsyslog_conf_files_output_modules:
+
+  - name: 'output-modules'
+    type: 'modules'
+    sections:
+
+      - comment: 'Log messages into files with timestamp format either traditional or default'
+        options: |-
+          {% if rsyslog_basics_use_traditional_timestamp_format|default(true) %}
+          module(load="builtin:omfile" Template="RSYSLOG_TraditionalFileFormat")
+          {% else %}
+          module(load="builtin:omfile")
+          {% endif %}
+
+# rsyslog_conf_files_output_rules:
+#
+# A set of default ``rsyslog`` options which store local system logs in files
+# located in `{{rsyslog_system_log_dir}}/` directory. This is mostly the
+# same as the default ``rsyslogd`` configuration provided in the installations.
+rsyslog_conf_files_output_rules:
+
+  - name: 'output-files'
+    type: 'output'
+    suffix: 'system'
+    path: '{{ rsyslog_config_dir }}'
+    sections:
+
+      - comment: |-
+          Log all kernel messages to the console.
+          Logging much else clutters up the screen.
+        options: |-
+          kern.*                                                 /dev/console
+        state: '{{ "present"
+                  if ("kernel-message" in rsyslog_capabilities)
+                  else "absent" }}'
+
+      - comment: |-
+          Log anything (except mail) of level info or higher.
+          Don't log private authentication messages!
+        options: |-
+          *.info;mail.none;authpriv.none;cron.none                {{rsyslog_system_log_dir}}/messages
```

Comment:
@nhosoi Why is each filter in a separate comment?

Reply:
Also, you will have here duplicates if:
"network" in rsyslog_capabilities and "remote-files" in rsyslog_capabilities.

Reply:
@sradco:
> @nhosoi Why is each filter in a separate comment?

That's only because I wanted to have the same comments in the original rsyslog.conf in the files_forwards output role.
```
#### RULES ####

# Log all kernel messages to the console.
# Logging much else clutters up the screen.
#kern.*                                                 /dev/console

# Log anything (except mail) of level info or higher.
# Don't log private authentication messages!
*.info;mail.none;authpriv.none;cron.none                /var/log/messages

....
```
> Also, you will have here duplicates for all the filters if:
> "network" in rsyslog_capabilities and "remote-files" in rsyslog_capabilities.

Could you elaborate?  I ran ansible-playbook with the following vars, but I only see one 30-output-files.system and there's no duplicates [1].
```
logging_enabled: true
rsyslog_capabilities: [ 'network', 'remote-files' ]
rsyslog_unprivileged: False
logging_purge_confs: True
rsyslog_backup_dir: /home/nhosoi/tmp/backup
logging_debug: True
logging_outputs:
  - name: files-and-forwards
    type: files_forwards
    logs_collections:
      - name: basics
        state: 'present'
```
[1]
```
# egrep '*.info;mail.none;authpriv.none;cron.none' *
30-output-files.system:*.info;mail.none;authpriv.none;cron.none                /var/log/messages
```

Reply:
> @sradco:
> 
> > @nhosoi Why is each filter in a separate comment?
> 
> That's only because I wanted to have the same comments in the original rsyslog.conf in the files_forwards output role.

@nhosoi I see. I thought you can have them all together as a single paragraph with the comments included.

> 
> ```
> #### RULES ####
> 
> # Log all kernel messages to the console.
> # Logging much else clutters up the screen.
> #kern.*                                                 /dev/console
> 
> # Log anything (except mail) of level info or higher.
> # Don't log private authentication messages!
> *.info;mail.none;authpriv.none;cron.none                /var/log/messages
> 
> ....
> ```
> 
> > Also, you will have here duplicates for all the filters if:
> > "network" in rsyslog_capabilities and "remote-files" in rsyslog_capabilities.
> 
> Could you elaborate? I ran ansible-playbook with the following vars, but I only see one 30-output-files.system and there's no duplicates [1].
> 
> ```
> logging_enabled: true
> rsyslog_capabilities: [ 'network', 'remote-files' ]
> rsyslog_unprivileged: False
> logging_purge_confs: True
> rsyslog_backup_dir: /home/nhosoi/tmp/backup
> logging_debug: True
> logging_outputs:
>   - name: files-and-forwards
>     type: files_forwards
>     logs_collections:
>       - name: basics
>         state: 'present'
> ```
> 
> [1]
> 
> ```
> # egrep '*.info;mail.none;authpriv.none;cron.none' *
> 30-output-files.system:*.info;mail.none;authpriv.none;cron.none                /var/log/messages
> ```
@nhosoi I thought you will have in both 30-remote-dynamic-logs.remote file and in 30-output-files.system.
In the remote use case when rsyslog_capabilities: [ 'network', 'remote-files' ], do you also need the 30-output-files.system?
How do you differentiate between the logs collected on the rsyslog server and the logs that are collected from the remote hosts in the rsyslog server?